src/nupic/algorithms/sdr_classifier.py

```diff
@@ -369,6 +363,90 @@ def inferSingleStep(self, patternNZ, weightMatrix):
     return predictDist
 
 
+  @classmethod
+  def read(cls, proto):
+    classifier = object.__new__(cls)
+
+    classifier.steps = []
+    for step in proto.steps:
```

Comment:
I couldn't do classifier.steps = proto.steps because proto.steps is of type "capnp list reader". So we have to unpack it here. I used list comparisons following your suggestions.